Sources/_SubprocessCShims/process_shims.c

```diff
@@ -664,6 +540,15 @@ int _subprocess_fork_exec(
         // If we reached this point, something went wrong
         write_error_and_exit;
     } else {
+        // On Linux 5.3 and lower, we have to fetch pidfd separately
+        // Newer Linux supports clone3 which returns pidfd directly
+        if (_pidfd < 0) {
+            _pidfd = _pidfd_open(childPid);
+            if (_pidfd < 0) {
+                return errno;
```

Comment:
We have a child pid though. I think we need to call waitid here before returning errno?

Reply:
Ahh good catch!